handlers/handlers_test.go

```diff
@@ -186,11 +250,76 @@ func defaultParams() url.Values {
 	values.Set("offset", "0")
 	values.Set("fromDate", "")
 	values.Set("toDate", "")
-	values.Set("sort", "release_date_desc")
+	values.Set("sort", queryparams.RelDateDesc.BackendString())
 	values.Set("keywords", "")
 	values.Set("query", "")
-	values.Set("release-type", "type-upcoming")
+	values.Set("release-type", queryparams.Upcoming.String())
 	values.Set("highlight", "true")
 
 	return values
 }
+
+func defaultICSParams() url.Values {
+	values := url.Values{}
+	values.Set("limit", "1000")
+	values.Set("toDate", time.Now().AddDate(0, 3, 0).Format(queryparams.DateFormat))
+	values.Set("sort", queryparams.RelDateAsc.BackendString())
+	values.Set("release-type", queryparams.Upcoming.String())
+
+	return values
+}
+
+func TestICalDate(t *testing.T) {
+	bad := []string{"1st Jan 2020", "21-03-2021", "2021-03-04T12:10:00"}
+	for _, d := range bad {
+		Convey("given a badly formatted date string "+d, t, func() {
+			Convey("then the iCalDate returns an empty string", func() {
+				icd := iCalDate(context.Background(), d)
+				So(icd, ShouldEqual, "")
+			})
+		})
+	}
+
+	good := []struct{ date, icalDate string }{
+		{date: "2021-03-04T12:10:00Z", icalDate: "20210304T121000Z"},
+		{date: "2021-03-04T12:10:00.000Z", icalDate: "20210304T121000Z"},
+		{date: "2021-03-04T12:10:00+05:00", icalDate: "20210304T071000Z"},
```

Comment:
Just thinking that we could simplify this test by adding the "bad" dates in here with an expected value of `""`?

Reply:
yeah could but not strictly necessary and quite like the breakdown between good and bad. But if you insist .... :-)